guix-science-nonfree/packages/nvidia-container-toolkit.scm

```diff
@@ -0,0 +1,203 @@
+		(("debug??libdir?") "debug")  ;ensure debug files get installed in the correct subdir
+		((".*nvidia-modprobe.mk.*") "\n")
+		(("^all: shared static tools")
+		 "all: shared tools")
+		((".*LIB_STATIC.*libdir.*$") ""))
+     	      (substitute* "mk/nvcgo.mk"
+		((".*-rf.*")
+		 "\tmkdir -p ${SRCS_DIR} && echo \"sources dir: ${SRCS_DIR}\"\n")
+		(("CURDIR./src/..PREFIX.")
+		 "CURDIR)/src/$(PREFIX)/*")) ;deleting sources fails
+              (substitute* "src/cli/libnvc.c"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
+              (substitute* "src/nvc_internal.h"
+		(("libnvidia-ml.so.1")
+		 "/run/current-system/profile/lib/libnvidia-ml.so.1"))
+	      (setenv "C_INCLUDE_PATH" (string-append (getenv "C_INCLUDE_PATH") ":" (string-append #$libtirpc "/include/tirpc")))
+     	      (setenv "LIBRARY_PATH" (string-append (getenv "LIBRARY_PATH") ":" (string-append #$libtirpc "/lib")))
+              (setenv "LDFLAGS" (string-append (or (getenv "LDFLAGS") "") " -ltirpc -lseccomp -lcap -Wl,-rpath=" (assoc-ref outputs "out") "/lib"))
+              (setenv "CFLAGS" (string-append (or (getenv "CFLAGS") "") " -DWITH_TIRPC -g"))
+       	      (substitute* "Makefile"
+		(("^WITH_LIBELF.*no")
+		 "WITH_LIBELF ?= yes"))
+    	      (substitute* "mk/common.mk"
+		(("^REVISION.*")
+		 (string-append "REVISION ?= " #$version "\n" "CC := gcc\n")))
+	      #true) ; must return true for success
+	    ))
+      #:tests? #f))
+    (native-inputs
+     (list libseccomp nvidia-modprobe which libtirpc libcap libelf git-minimal curl tar coreutils docker go gcc-toolchain rpcsvc-proto pkgconf))
+    (synopsis "Build and run containers leveraging NVIDIA GPUs")
+    (description "The NVIDIA Container Toolkit allows users to build and run GPU accelerated containers. The toolkit includes a container runtime library and utilities to automatically configure containers to leverage NVIDIA GPUs.")
+    (home-page "https://github.com/NVIDIA/nvidia-container-toolkit")
+    (license asl2.0)))
+
+
+(define-public nvidia-container-toolkit
+  (package
+    (name "nvidia-container-toolkit")
+    (version "1.13.1")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                    (url "https://github.com/NVIDIA/nvidia-container-toolkit")
+                    (commit (string-append "v" version))))
+              (file-name (git-file-name name version))
+              (sha256
+               (base32
+		"01gh57jfpcv07c4442lbf9wiy0l1iwl85ig9drpp0637gbkzgwa4"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/NVIDIA/nvidia-container-toolkit"
+      #:phases
+      #~(modify-phases %standard-phases
+  	  (add-after 'unpack 'fix-paths
+	    (lambda* (#:key inputs outputs #:allow-other-keys)
+	      (substitute* "src/github.com/NVIDIA/nvidia-container-toolkit/internal/config/config.go"
+(("/usr/bin")
+		 "/run/current-system/profile/bin"))))
```

Comment:
It would be best to refer to Coreutils' `/bin` (or something similar, depending on which executables are needed) than `/run/current-system/profile/bin`.

Reply:
it seems most of the binaries come from the package itself so I changed this to #$output/bin I'll test once I finish the rest of the edits so I'll leave this comment unresolved for now